### DISCUSSION TITLE
Despite default models have already been downloaded locally, why doesn't it work without network connection?

### DISCUSSION THREAD

**Discussion by cadake** [lean-dojo/LeanCopilot#101] [Q&A] [answered]
When I turn off the network connection, I got a mistake in vscode terminal which says "process 'git' exited with code 1"
![Snipaste_2024-07-21_22-56-32](https://github.com/user-attachments/assets/0a8452b9-a20a-4546-928c-1563aca143ec)

I have checked the models in cache in my wsl of ubuntu22 with copilot v1.4.1
![Snipaste_2024-07-21_22-56-32](https://github.com/user-attachments/assets/08f0a99c-7ea9-428c-b8f9-4c2bc1c2903d)

**Comment by Peiyang-Song** [comment, marked as answer]
@cadake Even after downloading the models to your local end, it is still required to have Internet, and specifically, Git access. This is because when running Lean Copilot's tactics, it will always automatically check whether the local models are the same as the latest remote models. If not, a warning will be displayed to suggest the users to download the latest models. This check would need access to our latest remote models hosted online, which needs the Internet.

**Reply by yangky11** [reply]
We could consider allowing users to disable this check via an environment variable.

**Reply by ZombiesR** [reply]
How can I do it by myself? I have same problem. Although I use vpn in china.
